sympy/utilities/iterables.py

```diff
@@ -2268,3 +2296,359 @@ def signed_permutations(t):
+    start = 0
+    while True:
+        i = find(seq, a, start)
+        if i == -1:
+            break
+        seq[i:i+len(a)] = b
+        start = i + len(b)
+
+
+def find(seq, subseq, start=0, all=False):
+    """Return starting index at which subsequence appears in sequence
+    at or beyond the index corresponding to position `start`, else -1.
+
+    Examples
+    ========
+
+    >>> from sympy.utilities.iterables import find
+    >>> find('alabaster', 'ba')
+    3
+    >>> find('alabaster', 'a', start=1)
+    2
+    >>> find('alabaster', 'e', start=-3)
+    7
+
+    To find all occurances at or past the start,
+    set keyword `all` to True:
+
+    >>> find('alabaster', 'a', all=True)
+    [0, 2, 4]
+    >>> find('alabaster', 'a', 1, all=True)
+    [2, 4]
+
+    See Also
+    ========
+    group, multiset, runs, split
+    """
+    # adapted from http://code.activestate.com/recipes/
+    # 117223-boyer-moore-horspool-string-searching/
+    from collections import defaultdict
+    m = len(subseq)
+    n = len(seq)
+    if m > n:
+        return -1 if not all else []
+    if start < 0:
+        start = max(-n, start) + n
+    skip = defaultdict(lambda: m)
+    for k in range(m - 1):
+        skip[subseq[k]] = m - k - 1
+    k = m - 1 + start
+    loc = []
+    while k < n:
+        j = m - 1
+        i = k
+        while j >= 0 and seq[i] == subseq[j]:
+            j -= 1
+            i -= 1
+        if j == -1:
+            if all:
+                loc.append(i + 1)
+            else:
+                return i + 1
+        k += skip[seq[k]]
+    return -1 if not all else loc
+
+
+def split(seq, ignore=[]):
+    """Split sequence into subsequences separated by elements that
+    are in ignore.
+
+    Examples
+    ========
+
+    >>> from sympy.utilities.iterables import split
+    >>> split('abbc', 'b')
+    ['a', 'c']
+
+    To split on a sequence, one can replace the sequence with a unique
+    element and then split on that element:
+
+    >>> from sympy.utilities.iterables import seq_replace
+    >>> from sympy.core.symbol import Dummy
+    >>> seq = list('abcskipd')
+    >>> uniq = Dummy()
+    >>> seq_replace(seq, list('skip'), uniq)
+    >>> split(seq, uniq)
+    [['a', 'b', 'c'], ['d']]
+
+    See Also
+    ========
+    find, group, multiset, seq_replace, runs
+    """
+    if not ignore:
+        return [seq]
+    if type(ignore) is not list and not isinstance(ignore, string_types):
+        ignore = [ignore]
+    rv = []
+    i = 0
+    ignoring = False
+    for j in range(len(seq)):
+        if seq[j] in ignore:
+            if ignoring:
+                i = j
+            else:
+                rv.append(seq[i:j])
+                ignoring = True
+        elif ignoring:
+            i = j
+            ignoring = False
+    if ignoring:
+        rv.append(seq[j:j])
+    else:
+        rv.append(seq[i:])
+    return rv
+
+
+def longest_repetition(*seq, **kwargs):
+    """Returns the longest, continguous repeated subsequence
```

Comment:
It is as intended: `xxx` is the longest string that is repeated; it's not trying to find the longest string with repeats. The non-overlapping special-handling portion would be easy to modify so that a keyword would be honored. Is that your preference?